lib/widgets/package_tile.dart

```diff
@@ -50,14 +49,20 @@ class PackageTile extends StatelessWidget {
               color: Colors.white,
             ),
           )),
-      onTap: () => Navigator.pushNamed(
-        context,
-        PackageDetailsPage.routeName,
-        arguments: PackageDetailsArguments(
-          page.packages[index].name,
-          page.packages[index].score.toString(),
-        ),
-      ),
+      onTap: () async {
+        final getPackageDetailsEvent = GetPackageDetailsEvent(
+            packageName: packageName, packageScore: packageScore);
+        BlocProvider.of<PackageDetailsBloc>(context)
+          ..dispatch(Initialize())..dispatch(getPackageDetailsEvent);
```

Comment:
why do you need to dispatch `Initialized()`? Shouldn't the bloc already be initialized when it's first created in BlocProvider?

Reply:
Initialize is really a call to get the first page. Since my bloc relies on async operations for data , I need my initial state to hold the first page. I can't do that with initial state. Any suggestions?

Reply:
that's fine but I would suggest moving the `..dispatch(Initialized())` to the BlocProvider that creates the `PackageDetailsBloc` like:
```dart
BlocProvider(
  builder: (context) => PackageDetailsBloc(...)..dispatch(Initialized()),
  child: ...
)
```

Reply:
OK, I'll take a look at that